File: app/src/main/java/org/neidhardt/dynamicsoundboard/views/recyclerviewhelpers/BaseAdapter.java
package org.neidhardt.dynamicsoundboard.views.recyclerviewhelpers;

import android.support.v7.widget.RecyclerView;
import org.neidhardt.dynamicsoundboard.mediaplayer.EnhancedMediaPlayer;

import java.util.List;

/**
 * File created by eric.neidhardt on 17.06.2015.
 */
public abstract class BaseAdapter<Type, ViewHolder extends RecyclerView.ViewHolder>
		extends
			RecyclerView.Adapter<ViewHolder>
		implements
			ListAdapter<Type>
{
	public abstract List<Type> getValues();

	@Override
	public void notifyItemChanged(Type data)
	{
		int index = this.getValues().indexOf(data);
		if (index == -1)
			this.notifyDataSetChanged();
		else
			this.notifyItemChanged(index);
	}
}

File: app/src/main/kotlin/org/neidhardt/dynamicsoundboard/views/recyclerviewhelpers/BaseAdapter.kt
package org.neidhardt.dynamicsoundboard.views.recyclerviewhelpers

import android.support.v7.widget.RecyclerView

/**
 * File created by eric.neidhardt on 17.06.2015.
 */
abstract class BaseAdapter<Type, ViewHolder : RecyclerView.ViewHolder> :
		RecyclerView.Adapter<ViewHolder>(),
		ListAdapter<Type>
{
	abstract val values: List<Type>

	override fun notifyItemChanged(data: Type)
	{
		val index = this.values.indexOf(data)
		if (index == -1)
			this.notifyDataSetChanged()
		else
			this.notifyItemChanged(index)
	}
}
